docs/source/rules.rst

```diff
@@ -33,14 +33,77 @@ Implementation
 .. autoclass:: sqlfluff.core.rules.base.LintFix
    :members:
 
+The `_eval` function of each rule takes a `RuleContext` parameter. It
+evaluate the given segment for violations, returning a `LintResult` if it
+finds an error. The `LintResult` includes a reference to the segment which
+"triggered" the error. Usually, it is the segment that needs correcting, **or**
+if the rule relates to something that is missing, then it should reference on
+the segment **following** the location where the missing element should be.
+
+Functional API
+--------------
+These newer modules provide a higher-level API for rules working with segments
+and slices. Rules that need to navigate or search the parse tree may benefit
+from using these. Eventually, the plan is for **all** rules to use these
+modules. At this time, the following rules use them:
+* `L001`
+* `L008`
+* `L009`
+* `L013`
+* `L015`
+* `L017`
+* `L021`
+* `L025`
+* `L031`
+* `L035`
+* `L036`
+* `L038`
+* `L042`
+* `L043`
+* `L047`
+* `L050`
+* `L052`
```

Comment:
I'm not sure it makes sense to keep this list given how many you've converted. It will just end up out of date. If we want to still give some measure of scale, but without having to maintain the list, it might be better to say something like: `...at 2021-12-30, 17 rules use this method.`

Reply:
Will do.